docs/getting_started/tutorial.rst

```diff
@@ -284,143 +443,402 @@ Adjust your blog page template to include the image:
+        <div class="tags">
+            <h3>Tags</h3>
+            {% for tag in page.specific.tags.all %}
+                <a href="{% slugurl 'tags' %}?tag={{ tag }}"><button type="button">{{ tag }}</button></a>
+            {% endfor %}
+        </div>
+    {% endif %}
 
-    {% load wagtailcore_tags %}
+Visiting a blog post with tags should now show a set of linked
+buttons at the bottom - one for each tag. However, clicking a button
+will get you a 404, since we haven't yet defined a "tags" view, which
+is going to require a little extra magic. Add to ``models.py:``
 
-    {% block body_class %}template-blogindexpage{% endblock %}
+.. code-block:: python
+
+    class BlogTagIndexPage(Page):
+
+        def get_context(self, request):
+
+            # Filter by tag
+            tag = request.GET.get('tag')
+            blogpages = BlogPage.objects.filter().filter(tags__name=tag)
+
+            # Update template context
+            context = super(BlogTagIndexPage, self).get_context(request)
+            context['blogpages'] = blogpages
+            return context
+
+Note that this Page-based model defines no fields of its own.
+Even without fields, subclassing ``Page`` makes it a part of the
+Wagtail ecosystem, so that you can give it a title and URL in the
+admin, and so that you can manipulate its contents by returning
+a queryset from its ``get_context()`` method.
+
+Migrate this in, then create a new ``BlogTagIndexPage`` in the admin.
+You'll probably want to create the new page/view under Homepage,
+parallel to your Blog index. Give it the slug "tags" on the Promote tab.
+
+Access ``/tags`` and Django will tell you what you probably already knew:
+you need to create a template ``blog/blog_tag_index_page.html:``
+
+.. code-block:: html+django
+
+    {% extends "base.html" %}
+    {% load wagtailcore_tags %}
 
     {% block content %}
-        <h1>{{ page.title }}</h1>
 
-        <div class="intro">{{ page.intro|richtext }}</div>
+        {% if request.GET.tag|length %}
+            <h4>Showing pages tagged "{{ request.GET.tag }}"</h4>
+        {% endif %}
+
+        {% for blogpage in blogpages %}
+
+              <p>
+                  <strong><a href="{% slugurl blogpage.slug %}">{{ blogpage.title }}</a></strong><br />
+                  <small>Revised: {{ blogpage.latest_revision_created_at }}</small><br />
+                  {% if blogpage.author %}
+                    <p>By {{ blogpage.author.profile }}</p>
+                  {% endif %}
+              </p>
+
+        {% empty %}
+            No pages found with that tag.
+        {% endfor %}
+
     {% endblock %}
 
-Related items
-~~~~~~~~~~~~~
+Unlike in the previous example, we're linking to pages here with the builtin ``slugurl``
+tag rather than ``pageurl``. The difference is that ``slugurl`` takes a Page slug
+(from the Promote tab) as an argument. ``pageurl`` is more commonly used because it
+is unambiguous, but use whichever one best suits your purpose.
```

Comment:
Sure, if you think that works better - although then that leaves the question of how to do the link to BlogTagIndexPage. (Having said that, simply hard-coding a link to `/tags/` _is_ a valid option.) If you're not certain, feel free to leave this one with me...